src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx

```diff
@@ -37,59 +35,41 @@ export interface RegisterData {
   date_of_birth: string;
 }
 
-const CODE_OF_CONDUCT_QUESTIONS: CodeOfConductQuestion[] = [
-  {
-    name: "termsAndConditions",
-    text: "I agree to SparkleVerse's Terms and Conditions",
-    link: "https://sparklever.se/terms-and-conditions",
-  },
-  {
-    name: "commonDecency",
-    text:
-      "I will endeavor not to create indecent experiences or content, and understand my actions may be subject to review and possible disciplinary action",
-  },
-  {
-    name: "regionalBurn",
-    text:
-      "I understand this is an Australian Regional Burn guided by the Ten Principles of Burning Man",
-  },
-];
-
 const RegisterForm: React.FunctionComponent<PropsType> = ({
   displayLoginForm,
   displayPasswordResetForm,
   afterUserIsLoggedIn,
   closeAuthenticationModal,
 }) => {
   const history = useHistory();
-  const venue = useSelector((state) => state.firestore.ordered.currentVenue);
+  const venue = useSelector(
+    (state) => state.firestore.ordered.currentVenue?.[0]
+  );
 
   const signUp = ({ email, password }: RegisterFormData) => {
     return firebase.auth().createUserWithEmailAndPassword(email, password);
   };
 
-  const {
-    register,
-    handleSubmit,
-    errors,
-    formState,
-    setError,
-    watch,
-  } = useForm<RegisterFormData>({
+  const { register, handleSubmit, errors, formState, setError } = useForm<
+    RegisterFormData
+  >({
     mode: "onChange",
   });
 
   if (!venue) {
     return <>Loading...</>;
   }
 
-  const venueId = venue[0]?.id;
-
   const onSubmit = async (data: RegisterFormData) => {
     try {
-      if (CODE_CHECK_ENABLED) await axios.get(codeCheckUrl(data.code));
+      if (venue.requiresTicketCode) await axios.get(codeCheckUrl(data.code));
+      if (venue.requiresEmailVerification)
+        await axios.get(codeCheckUrl(data.email));
       const auth = await signUp(data);
-      if (CODE_CHECK_ENABLED && auth.user) {
+      if (
+        auth.user &&
+        (venue.requiresTicketCode || venue.requiresEmailVerification)
+      ) {
```

Comment:
[DRY] This seems the same as another block of similar code. Can DRY this up in future.